### PR TITLE
8272216: G1: replace G1ParScanThreadState::_dest with a constant

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple change of removing an array with statically-known identical elements.

Test: hotspot_gc

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272216](https://bugs.openjdk.java.net/browse/JDK-8272216): G1: replace G1ParScanThreadState::_dest with a constant

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to a134b43260579bf7c9735d10ca24e7821d5daa9d
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5065/head:pull/5065` \
`$ git checkout pull/5065`

Update a local copy of the PR: \
`$ git checkout pull/5065` \
`$ git pull https://git.openjdk.java.net/jdk pull/5065/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5065`

View PR using the GUI difftool: \
`$ git pr show -t 5065`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5065.diff">https://git.openjdk.java.net/jdk/pull/5065.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1ParScanThreadState.cpp [hunk 1]
```diff
@@ -363,7 +358,9 @@ G1HeapRegionAttr G1ParScanThreadState::next_region_attr(G1HeapRegionAttr const r
       return region_attr;
     }
   }
-  return dest(region_attr);
+  assert(region_attr.is_young() || region_attr.is_old(), "must be either Young or Old");
```
Comment (kimbarrett) [inline src/hotspot/share/gc/g1/g1ParScanThreadState.cpp:361]: I think I'd rather have this assert at the beginning of the function.

**Review by kimbarrett — APPROVED** [review]
Looks good.  Just a suggested change to assert placement.

File: src/hotspot/share/gc/g1/g1ParScanThreadState.cpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/g1/g1ParScanThreadState.cpp]: Revised.

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**